types/TestResult.js

```diff
@@ -105,14 +139,36 @@ export type TestResult = {|
   testResults: Array<AssertionResult>,
 |};
 
+export type FormattedTestResult = {|
+  message: string,
+  name: string,
+  summary: string,
+  status: 'failed' | 'passed',
+  startTime: number,
+  endTime: number,
+  coverage: any,
+  assertionResults: Array<FormattedTestAssertion>,
+|};
+
+export type FormattedTestResults = AggregatedResult & {
+  numFailedTests: number,
```

Comment:
Good spot, I can simplify this further, I think I might initially took the result of formatTestResults and typed around it, since I've tried different solutions (thanks to @jeanlauliac for a lot of suggestions here), when removing unnecessary types I probably missed this one.